bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/Storage.java

```diff
@@ -1778,37 +1762,51 @@ private Properties findVMProfile(Generation systemGeneration) {
 		return result;
 	}
 
+	private InputStream createProfileStream(Generation systemGeneration, String vmProfile,
+			String embeddedProfileName) {
+		InputStream profileIn = null;
+		if (vmProfile != null) {
+			// look for a profile in the system bundle based on the vm profile
+			String javaProfile = vmProfile + PROFILE_EXT;
+			profileIn = findInSystemBundle(systemGeneration, javaProfile);
+			if (profileIn == null)
+				profileIn = getNextBestProfile(systemGeneration, JAVASE, runtimeVersion, embeddedProfileName);
+		}
+		if (profileIn == null)
+			// the profile url is still null then use the min profile the framework can use
+			profileIn = findInSystemBundle(systemGeneration, "JavaSE-1.8.profile"); //$NON-NLS-1$
+		return profileIn;
+	}
```

Comment:
I think this is a good opportunity to add some braces :)
```suggestion
	private InputStream createProfileStream(Generation systemGeneration, String vmProfile, String embeddedProfileName) {
		InputStream profileIn = null;
		if (vmProfile != null) {
			// look for a profile in the system bundle based on the vm profile
			profileIn = findInSystemBundle(systemGeneration, vmProfile + PROFILE_EXT);
			if (profileIn == null) {
				profileIn = getNextBestProfile(systemGeneration, JAVASE, runtimeVersion, embeddedProfileName);
			}
		}
		if (profileIn == null) {
			// the profile url is still null then use the min profile the framework can use
			profileIn = findInSystemBundle(systemGeneration, "JavaSE-1.8.profile"); //$NON-NLS-1$
		}
		return profileIn;
	}
```